Graphs/prim.cpp

```diff
@@ -0,0 +1,74 @@
+// Prim's Algorithm is used to find the Minimum Spanning Tree (MST) of a graph, 
+// minimizing the total edge weight while connecting all vertices without cycles.
+
+#include <iostream>
+#include <vector>
+#include <queue>
+#include <utility>
+#include <functional>
+
+using namespace std;
+
+struct PrimAlgorithm {
+    int n; // number of vertices
+    vector<vector<pair<int, int>>> adj;
+
+    PrimAlgorithm(int n) : n(n), adj(n) {}
+
+    void addEdge(int u, int v, int weight) {
+        adj[u].push_back(make_pair(v, weight));
+        adj[v].push_back(make_pair(u, weight));
+    }
+
+    int minimumSpanningTree(int start = 0) {
+        vector<bool> inMST(n, false);
+        vector<int> minEdge(n, 999999);
+        minEdge[start] = 0;
+        priority_queue<pair<int, int>, vector<pair<int, int>>, greater<pair<int, int>>> pq;
+
+        pq.push(make_pair(0, start));
+
+        int mstWeight = 0;
+
+        while (!pq.empty()) {
+            int u = pq.top().second;
+            int weight = pq.top().first;
+            pq.pop();
+
+            if (inMST[u]) continue;
+
+            inMST[u] = true;
+            mstWeight += weight;
+
+            for (auto &neighbor : adj[u]) {
+                int v = neighbor.first;
+                int w = neighbor.second;
```

Comment:
This can be simplified as such:

```
for (auto [v, w]: adj[u]) {
....
```

and the lines 44 and 45 `(int u, int v)` won't longer be needed.